challenge-3.js

```diff
@@ -0,0 +1,42 @@
+function calcAge(birthDate) {
+  if (!isDateValid(birthDate)) return "bad request!";
+  const birthDateExtraction = {
+    year: birthDate.getYear(),
+    month: birthDate.getMonth(),
+    day: birthDate.getDate(),
+  };
+  let now = new Date();
+  const currentDate = {
+    year: now.getYear(),
+    month: now.getMonth(),
+    day: now.getDate(),
+  };
+  const age = {
+    year: currentDate.year - birthDateExtraction.year,
+    month: currentDate.month - birthDateExtraction.month,
+    day: currentDate.day - birthDateExtraction.day,
+  };
+  if (age.month < 0) {
+    age.month += 12;
+    age.year--;
+  }
+  if (age.day < 0) {
+    age.day += 31;
+    age.month--;
+  }
+  return age;
+}
+
+const isDateValid = function (date) {
+  return date instanceof Date && date <= new Date();
```

Comment:
Keep in mind that this function is responsible for doing two things at a time. 

- Checking the date has the correct type
- Checking if the given date is older than right now